otel.go

```diff
@@ -12,41 +13,50 @@ import (
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
+	otelTrace "go.opentelemetry.io/otel/trace"
+	"sync"
 )
 
 const (
-	packageName = "github.com/evergreen-ci/logkeeper"
+	packageName = "github.com/evergreen-ci/logkeeper/%s"
+	defaultName = "noop_tracer"
 )
 
-func (lk *logkeeper) initOtel(ctx context.Context) error {
-	if lk.opts.TraceCollectorEndpoint == "" {
-		// defaults to NoopTracerProvider
-		lk.tracer = otel.GetTracerProvider().Tracer(packageName)
-		return nil
-	}
+var (
+	closers       []closerOp
+	creator       sync.Once
+	closer        sync.Once
+	useCustomName bool
+)
 
+func LoadTraceProvider(ctx context.Context, useInsecure bool, collectorEndpoint string, sampleRatio float64) {
+	if collectorEndpoint == "" {
+		return
+	}
 	r, err := serviceResource(ctx)
 	if err != nil {
-		return errors.Wrap(err, "making host resource")
+		grip.EmergencyFatal(errors.Wrap(err, "making host resource"))
 	}
-
-	lk.otelGrpcConn, err = grpc.DialContext(ctx,
-		lk.opts.TraceCollectorEndpoint,
-		grpc.WithTransportCredentials(credentials.NewTLS(nil)),
-	)
-	if err != nil {
-		return errors.Wrapf(err, "opening gRPC connection to '%s'", lk.opts.TraceCollectorEndpoint)
+	var opts []otlptracegrpc.Option
+	if useInsecure {
+		opts = append(opts, otlptracegrpc.WithInsecure())
 	}
+	opts = append(opts, otlptracegrpc.WithEndpoint(collectorEndpoint))
+	client := otlptracegrpc.NewClient(opts...)
 
-	client := otlptracegrpc.NewClient(otlptracegrpc.WithGRPCConn(lk.otelGrpcConn))
 	traceExporter, err := otlptrace.New(ctx, client)
 	if err != nil {
-		return errors.Wrap(err, "initializing otel exporter")
+		grip.EmergencyFatal(errors.Wrap(err, "initializing otel exporter"))
 	}
 	tp := sdktrace.NewTracerProvider(
 		sdktrace.WithBatcher(traceExporter),
+		sdktrace.WithSampler(
```

Comment:
Do we need to do sampling in the app? Is the concern that we'll overwhelm the collector?
Separately, won't we always have the parent span since the entire trace originates on this one app server?

Reply:
I set the sampler to make sampling decisions based on the parent span by default. If the remote/local parent span is being sampled, the sampler will also sample the span in `logkeeper`. If there isn't a parent span, the sampler will use the preset ratio for sampling.

Reply:
But when will the parent ever not be sampled?